pisa/analysis/analysis.py

```diff
@@ -619,15 +619,25 @@ def get_detailed_metric_info(data_dist, hypo_asimov_dist, params, metric,general
                 detailed_metric_info[m] = name_vals_d
 
             else:
-                if isinstance(hypo_asimov_dist,OrderedDict):
+                if isinstance(hypo_asimov_dist, OrderedDict):
                     hypo_asimov_dist = hypo_asimov_dist['weights']
 
-                name_vals_d['maps'] = data_dist.metric_per_map(
-                    expected_values=hypo_asimov_dist, metric=m
-                )
-                metric_hists = data_dist.metric_per_map(
-                    expected_values=hypo_asimov_dist, metric='binned_'+m
-                )
+                if m == 'weighted_chi2':
+                    actual_values = data_dist.hist['total']
+                    expected_values = hypo_asimov_dist.hist['total']
+                    d = {'output_binning': hypo_maker.pipelines[0].output_binning,
+                         'output_key': 'bin_unc2'}
```

Comment:
Just so i understand: `bin_unc2` is something that is provided at some point earlier in the pipeline for this to work, right?

Reply:
Correct, that is calculated in `pisa/stages/utils/hist.py` based on per event uncertainties provided to the pipeline